src/test/regress/sql/multi_mx_function_call_delegation.sql

```diff
@@ -1,69 +1,147 @@
+END;$$;
+
+-- Test that undistributed functions have no issue executing
+select mx_call_func(2, 0);
+select mx_call_func_custom_types('S', 'A');
+
+-- Same for unqualified names
+select mx_call_func(2, 0);
+select mx_call_func_custom_types('S', 'A');
+
+-- Mark both functions as distributed ...
+select create_distributed_function('mx_call_func(int,int)');
+select create_distributed_function('mx_call_func_custom_types(mx_call_enum,mx_call_enum)');
 
--- Test that undistributed procedures have no issue executing
+-- We still don't route them to the workers, because they aren't
+-- colocated with any distributed tables.
+SET client_min_messages TO DEBUG1;
 select mx_call_func(2, 0);
+select mx_call_func_custom_types('S', 'A');
 
--- Test some straight forward distributed calls
-select create_distributed_function('mx_call_add(int,int)');
-select create_distributed_function('mx_call_func(int,int)', 'x', colocate_with := 'mx_call_dist_table');
-select create_distributed_function('mx_call_func_raise(int)', '$1');
+-- Mark both functions as distributed ...
+select create_distributed_function('mx_call_func(int,int)');
+select create_distributed_function('mx_call_func_custom_types(mx_call_enum,mx_call_enum)');
 
-set client_min_messages to DEBUG;
+-- We still don't route them to the workers, because they aren't
+-- colocated with any distributed tables.
+SET client_min_messages TO DEBUG1;
+select mx_call_func(2, 0);
+select mx_call_func_custom_types('S', 'A');
 
+-- Mark them as colocated with a table. Now we should route them to workers.
+select colocate_func_with_table('mx_call_func', 'mx_call_dist_table_1'::regclass, 1);
+select colocate_func_with_table('mx_call_func_custom_types', 'mx_call_dist_table_enum'::regclass, 1);
 select mx_call_func(2, 0);
-select mx_call_func_raise(2);
+select mx_call_func_custom_types('S', 'A');
 
 -- We don't allow distributing calls inside transactions
 begin;
 select mx_call_func(2, 0);
 commit;
 
+-- Drop the table colocated with mx_call_func_custom_types. Now it shouldn't
+-- be routed to workers anymore.
+SET client_min_messages TO NOTICE;
+drop table mx_call_dist_table_enum;
+SET client_min_messages TO DEBUG1;
+select mx_call_func_custom_types('S', 'A');
+
 -- Make sure we do bounds checking on distributed argument index
 -- This also tests that we have cache invalidation for pg_dist_object updates
-select mx_call_dist_object('mx_call_func', 'mx_call_dist_table'::regclass, -1);
+select colocate_func_with_table('mx_call_func', 'mx_call_dist_table_1'::regclass, -1);
 select mx_call_func(2, 0);
-select mx_call_dist_object('mx_call_func', 'mx_call_dist_table'::regclass, 2);
+select colocate_func_with_table('mx_call_func', 'mx_call_dist_table_1'::regclass, 2);
 select mx_call_func(2, 0);
-select mx_call_dist_object('mx_call_func', 'mx_call_dist_table'::regclass, 1);
 
--- test non Const distribution parameter
-select mx_call_func(2, mx_call_add(3, 4));
```

Comment:
Also add these, similar to https://github.com/citusdata/citus/pull/3032/files, but also make sure the behaviour is as we expect.